File: 310971/Hrakovich/src/main/java/org/example/NoticeMapper.java
package org.example;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface NoticeMapper {

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "story", ignore = true)
    Notice toEntity(NoticeRequestTo dto);

    @Mapping(target = "storyId", source = "story.id")
    NoticeResponseTo toResponse(Notice entity);
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/mapper/NoticeMapper.kt
package com.distcomp.mapper

import com.distcomp.dto.notice.NoticeRequestTo
import com.distcomp.dto.notice.NoticeResponseTo
import com.distcomp.entity.Notice
import org.mapstruct.Mapper
import org.mapstruct.MappingConstants
import org.mapstruct.ReportingPolicy

@Mapper(
    componentModel = MappingConstants.ComponentModel.SPRING,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface NoticeMapper {
    fun toNoticeResponse(notice: Notice) : NoticeResponseTo

    fun toNewsEntity(noticeRequestTo: NoticeRequestTo) : Notice
}